mev-boost-rs/src/relay_mux.rs

```diff
@@ -132,12 +132,14 @@ impl BlindedBlockProvider for RelayMux {
             return Err(Error::NoBids)
         }
 
-        // for now, break any ties by picking the first bid,
-        // which currently corresponds to the fastest relay
-        let (best_index, rest) = best_indices.split_first().unwrap();
+        // If multiple indices with same bid value break tie by selecting largest Execution Block Hash
+        let best_index = best_indices
+            .iter()
+            .max_by(|&x, &y| bids[*x].0.block_hash().cmp(bids[*y].0.block_hash()))
```

Comment:
after thinking about this some, we _could_ just randomly pick one, rather than use the block hash as the key to break ties -- this would remove any incentive to "grind" a particular hash for your payload (and again, the likelihood of builders doing this is _super_ low)

using the block hash as entropy here is only important if we need non-determinism but we don't here, and in fact removes the prior incentives if we have it

that being said, I'm fine to leave this as is for now unless you want to go and just use `rand` to just grab an index from this list